src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -268,23 +268,38 @@ public static void Move(string sourceDirName, string destDirName)
             string fulldestDirName = Path.GetFullPath(destDirName);
             string destPath = PathInternal.EnsureTrailingSeparator(fulldestDirName);
 
-            StringComparison pathComparison = PathInternal.StringComparison;
+            // Here we have a bool to check if they're equal, but with different casing
+            // We allowing moving foo to the same directory, but renaming with a different case.
+            bool sameDirectoryIgnoreCase =
+                string.Equals(sourcePath, destPath, StringComparison.OrdinalIgnoreCase);
 
-            if (string.Equals(sourcePath, destPath, pathComparison))
+            StringComparison pathComparison = StringComparison.Ordinal;
+
+            // If the paths are the exact same, fail.
+            if (sameDirectoryIgnoreCase
+                && string.Equals(sourcePath, destPath, pathComparison))
+            {
                 throw new IOException(SR.IO_SourceDestMustBeDifferent);
+            }
+
 
             string sourceRoot = Path.GetPathRoot(sourcePath);
             string destinationRoot = Path.GetPathRoot(destPath);
             if (!string.Equals(sourceRoot, destinationRoot, pathComparison))
```

Comment:
You should add tests for roots that differ by case. Moving `C:\foo` to `c:\bar` should not fail and presumably it does now.